src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/IntervalTracker.cs

```diff
@@ -19,12 +19,29 @@ public class IntervalTracker
         private readonly Stopwatch _intervalWatch = new Stopwatch();
         private readonly List<Tuple<string, TimeSpan>> _intervalList = new List<Tuple<string, TimeSpan>>();
         private readonly string _activityName;
+        private readonly bool _isPopulatingIntervalList;
 
-        public IntervalTracker(string activityName)
+        /// <summary>
+        /// A tracker which creates <see cref="Interval"/> objects and emits them as ETW events with or without keeping references
+        /// for future usages.
+        /// </summary>
+        /// <param name="activityName"></param>
+        /// <param name="isPopulatingIntervalList">True to keep references to all intervals</param>
+        public IntervalTracker(string activityName, bool isPopulatingIntervalList)
```

Comment:
Would it make sense to refactor interval instead to allow you to fire ETW events instead, since that's what you need. You're not really tracking the intervals here.

Reply:
I think my point is that you don't need a tracker, which is just a wrapper around many Interval instances, but rather you can probably use the Interval instances themselves.